lib/dpl/provider/cloud_foundry.rb

```diff
@@ -23,8 +23,10 @@ def needs_key?
       end
 
       def push_app
-        context.shell "./cf push#{manifest}"
-        context.shell "./cf logout"
+        error 'Failed to push app' unless context.shell("./cf push#{manifest}")
+
+        ensure
```

Comment:
Stylistically, this line and the line below should be one level up (delete 2 spaces at the beginning of the lines).